test/bats/test.bats

```diff
@@ -78,10 +78,6 @@ SLEEP_TIME=1
   assert_success
 }
 
-@test "waiting for namespaces to be synced" {
```

Comment:
We need some way of testing whether the namespace has been synced to make the e2e tests a bit less flaky. Maybe we can create a special constraint on configmaps or namespaces that fails once the namespace is synced?

Another option may be to add annotations demonstrating sync status, but that would require resurrecting the cleanup logic removed in this PR.